modules/code-deployment/src/main/java/org/apache/ignite/internal/deployunit/message/UndeployUnitResponse.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.deployunit.message;
+
+import org.apache.ignite.network.NetworkMessage;
+import org.apache.ignite.network.annotations.Marshallable;
+import org.apache.ignite.network.annotations.Transferable;
+
+/**
+ * Undeploy unit response.
+ */
+@Transferable(DeployUnitMessageTypes.UNDEPLOY_UNIT_RESPONSE)
+public interface UndeployUnitResponse extends NetworkMessage {
+
+    /**
+     * Returns error which happens on undeploy process.
+     *
+     * @return error which happens on undeploy process.
+     */
+    @Marshallable
+    Throwable error();
```

Comment:
This one is interesting. I'm not sure that we should send entire exception instances. How is it done in other places?

Reply:
I don't think that I understand the answer. Throwable objects are quite big, so I'm interested, do we really need to send them entirely?